ckan/lib/app_globals.py

```diff
@@ -70,6 +70,7 @@
     'ckan.user_list_limit': {'default': '20', 'type': 'int'},
     'search.facets.default': {'default': '10', 'type': 'int',
                              'name': 'facets_default_number'},
+    'search_facets_limits': {'default': 3},
```

Comment:
i will remove this but im not sure why im getting that error

Reply:
@amercader im not able to start the app without this line, any suggestions ?

Reply:
As i understand, this variable is required by [search snippet](https://github.com/ckan/ckan/blob/master/ckan/templates/home/snippets/search.html#L1). So, if we are going to avoid adding global variables - we need to modify `get_facet_items_dict` helper

Reply:
Exactly, this [helper](https://github.com/tino097/ckan/blob/d5917bc8fc56498a503d07b5145e2916355bfbc8/ckan/lib/helpers.py#L931) is using `c.search_facets_limit`, which in the old Pylons based globals failed silently to an empty string so the helper didn't fail.

It looks like this property is actually set by the package and group controllers to track some state so it can't be moved to `config`:

```
ag search_facets_limits
ckan/lib/helpers.py
931:    if c.search_facets_limits and limit is None:
932:        limit = c.search_facets_limits.get(facet)
963:    if c.search_facets_limits and limit is None:
964:        limit = c.search_facets_limits.get(facet)

ckan/controllers/group.py
358:            c.search_facets_limits = {}
362:                c.search_facets_limits[facet] = limit

ckan/controllers/package.py
298:        c.search_facets_limits = {}
307:            c.search_facets_limits[facet] = limit
```

In this case lets do `hasattr(c, 'search_facets_limit')` on the helper to avoid the exception and not add it to the globals object by default. Does this make sense?

Also worth reading https://github.com/ckan/ckan/pull/3203 for context on `c` / `g`.

Reply:
can we not pass the filter information as a normal parameter instead?

Reply:
It is trickier than it seems: https://gist.github.com/amercader/ce4cae4a789a7db6fb78d304164e552a, and I didn't manage to make it work on the group controller because of how it is structured. Also there are a ton of other search and facet related variables in `c`, it might be best to refactor the whole thing consistently and not individual vars at this point.

Reply:
yes, that makes sense to me